### PR TITLE
[BEEEP]  Bitwarden Script uninstall option

### PR DESCRIPTION
**Pull request by mimartin12**
## Type of change
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Adds an uninstall option for both the Bash and PowerShell scripts for self hosted installation. This allows users to cleanly remove the Bitwarden self hosted stack from the system.

## Code changes

* **scripts/bitwarden.ps1:** Added "uninstall" switch so it's allowed to be passed in as `./bitwarden.ps1 -uninstall`
* **scripts/bitwarden.sh:** Added "uninstall" switch so it's allowed to be passed in as `./bitwarden.sh uninstall`
* **scripts/run.ps1:** Added Uninstall() function, it warns user that this process will delete data, then tears down docker containers, removes the `bwdata` folder, and prompts to prune docker images.
* **scripts/run.sh:** Added uninstall() function, it warns user that this process will delete data, then tears down docker container, removes the `bwdata` folder, removes docker volumes, and then prompts to prune docker images.
* **util/Setup/Templates/DockerCompose.hbs:** Adds support to label volumes with a Bitwarden label so that when removing volumes, we can target the specific volume. 

## Testing requirements

This needs standard self-hosted setup testing. The expectation is that after install Bitwarden, a user can use either `./bitwarden.sh uninstall` or `./bitwarden.ps1 -uninstall` to remove all remnants of the self hosted Bitwarden from their system. 

### Requirements
- git
- docker
- docker-compose
- dotnet 5.0

#### Clean install

1. Set up a clean install of the `dev` self-host server

####  Build new Setup image

1. clone `server` repo
2. `git switch beeep/uninstall-option`
3. `cd server/util/Setup`
4. `./build.sh && docker build -t bitwarden/setup:dev .`

#### Edit scripts to use new Setup image
With the current way we are releasing the self host installation scripts, we need to make some modifications so that we don't pull the latest released `bitwarden/setup:dev` image. These are the original self-host install scripts and not their respective counterparts in the newly cloned `server` project.

1. edit bitwarden.sh
  a. change COREVERSION to latest
  b. add a return to the top of the downloadRunFile function
    ```
    function downloadRunFile() {
      return
      ...
    }
    ```
2. edit bwdata/scripts/run.sh
  a. add a return line to the pullSetup function
    ```
    function pullSetup() {
      return
      ...
    }
    ```

#### Install Bitwarden and test uninstall

1. Run a `./bitwarden.sh install`
2. Follow all the prompts.
3. Once installed, start Bitwarden `./bitwarden.sh start`
4. Uninstall Bitwarden `./bitwarden.sh uninstall`
5. Enter `y` for all the prompts.
6. Validate the following is true
     1. No Bitwarden docker containers are running.
     2. No Bitwarden docker volumes exist.
     3. No Bitwarden container images exist.
     4. `bwdata` no longer exists. 

## Before you submit
- [X] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by vgrassia** [comment]
Have these been tested?

- Uninstalling an old installation
- Upgrading an old installation, then uninstalling

**Comment by mimartin12** [comment]
> Have these been tested?
> 
> * Uninstalling an old installation
> * Upgrading an old installation, then uninstalling

I have not tested those scenarios, I can do that now. How old of a version are you thinking? It shouldn't interfere with anything since these uninstall functions really are just using the already functioning `dockerPurge`, `dockerComposeDown`, and removal of folders.

**Comment by mimartin12** [comment]
> Have these been tested?
> 
> * Uninstalling an old installation
> * Upgrading an old installation, then uninstalling

I can confirm after testing that the uninstall will work. Here is what I did to complete the tests.

1. Install an old version of Bitwarden. (1.45.0 core, 1.24.0 web)
2. Run the uninstall command, uninstall completed without issues.

1. Install an older version of Bitwarden. (1.45.0 core, 1.24.0 web)
2. Update Bitwarden to the latest version (1.45.2 core, 2.25.0 web)
3. Run the uninstall command, uninstall completed without issues.

**Comment by vgrassia** [comment]
I think the code looks good.  My concerns are with removing the Docker volume and the backups in the `mssql` folder.  As someone who self-hosts I would never want an uninstall to remove backups as well, I'd remove those manually if needed. I'd like to hear what @joseph-flinn thinks.

**Comment by mimartin12** [comment]
> 

Indeed, I was thinking about that as well. I think that if the concern is loosing data, then an additional prompt would need to be added. Something that allows the user to uninstall Bitwarden, but leave the backups and `bwdata` folder OR completely remove Bitwarden from their system.

**Comment by joseph-flinn** [comment]
I like the idea of completely purging a system of the `bwdata` folder. This would make testing the `bitwarden.sh` and `run.sh` changes a lot more streamlined. To mitigate Vince's concern, maybe we can prompt the user to either manually move the backups out of the `bwdata/mssql` directory or provide an automated path to put them in the directory alongside the `bitwarden.sh` script?

**Review by joseph-flinn — DISMISSED** [review]
Approved! Let's wait until Thursday with the server update to merge.

**Comment by vgrassia** [comment]
Do most Linux distros include `zip` by default?  I think `tar` might be more popular. Thoughts?

File: scripts/run.sh
Comment (vgrassia) [inline scripts/run.sh:239]:
```suggestion
        tar -cvzf "./bitwarden_database.tar.gz" "$OUTPUT_DIR/mssql"
```

Comment (vgrassia) [reply scripts/run.sh:239]: Do most Linux distros include zip by default? I think tar might be more popular. Thoughts?

**Review by vgrassia — APPROVED** [review]

**Pull request merged**
